### DISCUSSION TITLE
Image Upload Failed

### DISCUSSION THREAD

**Discussion by pranaykotas** [outline/outline#3521] [Self Hosting]
I'm unable to upload images to AWS S3. I read the other discussions and set `AWS_S3_ACL=public-read`

However, it still results in an 'upload failed' error.

Relevant lines from the docker-compose.yml file are given below. What am I doing wrong?

      - AWS_ACCESS_KEY_ID=something
      - AWS_SECRET_ACCESS_KEY=something
      - AWS_REGION=us-east-1
      - AWS_S3_UPLOAD_BUCKET_URL=http://outline-something.s3-website-us-east-1.amazonaws.com
      - AWS_S3_UPLOAD_BUCKET_NAME=outline-something
      - AWS_S3_UPLOAD_MAX_SIZE=26214400
      ##- AWS_S3_FORCE_PATH_STYLE=false
      - AWS_S3_FORCE_PATH_STYLE=true
      - AWS_S3_ACL=public-read
      - PGSSLMODE=disable

Maybe I'm doing something wrong in the Amazon s3 settings?

On the S3 console, I've added the CORS policy mentioned [here](https://app.getoutline.com/share/770a97da-13e5-401e-9f8a-37949c19f97e/doc/aws-s3-N4M0T6Ypu7). 
“Block public access” is set to “off” at the bucket level.
I've enabled static website hosting but it is set to "host a specific website" option.

I'm a complete noob to web development so please forgive for the dumb questions.

**Comment by almereyda** [comment]
Do you see something correlated to this error message in the network tab of
your browser's developer tools?

**Reply by pranaykotas** [reply]
Thanks for your response. Yes, I do see this message:
`POST https://outline-something.s3-website-us-east-1.amazonaws.com/ net::ERR_CONNECTION_TIMED_OUT`

When I go to the Network tab in developer tools by clicking on the https link above, it says 'Failed to load response data: no data found for resource with given identifier'.

**Comment by almereyda** [comment]
It seems this could be some kind of network split, eventually caused by
hard blocking of third party origins, or an upstream issue at AWS. Would
you find other reasons in your local network setup and your browser, why
this connection could time out?

**Comment by heyvaldemar** [comment]
hey guys, I wrote [a detailed guide](https://www.heyvaldemar.com/install-outline-and-keycloak-using-docker-compose/) after I was able to configure Outline with MinIO and Keycloak.

Hopefully, it will help others.
